Tests/ValidationTests.swift

```diff
@@ -756,18 +756,19 @@ class AutomaticValidationTestCase: BaseTestCase {
 
 private enum ValidationError: Error {
     case missingData, missingFile, fileReadFailed
+    var error: AFError { return .responseValidationFailed(reason: .customValidationFailed(error: self)) }
 }
 
 extension DataRequest {
     func validateDataExists() -> Self {
         return validate { request, response, data in
-            guard data != nil else { return .failure(ValidationError.missingData) }
+            guard data != nil else { return .failure(ValidationError.missingData.error) }
             return .success(Void())
         }
     }
 
     func validate(with error: Error) -> Self {
-        return validate { _, _, _ in .failure(error) }
+        return validate { _, _, _ in .failure(error as? AFError ?? .responseValidationFailed(reason: .customValidationFailed(error: error))) }
```

Comment:
Yeah, I think leaving `ValidationResult` alone will simplify this usage, it just needs an internal cast when getting validation errors.

Reply:
Reverted.